server/server.js

```diff
@@ -24,10 +27,63 @@ app.get("/api/rentals", (req, res) => {
     });
 });
 
-app.get("/app.js", (req, res) => {
-  res.sendFile(path.resolve(__dirname, "../public/bundle.js"));
+app.delete("/api/rentals", (req, res) => {
+  const id = parseInt(req.query.id);
+
+  Rental.deleteOne({ _id: id })
+    .then(result => {
+      let message;
+      if (result.deletedCount === 0) {
+        message = `No rental id ${id} to delete`;
+      } else {
+        message = `Rental id ${id} deleted`
+      }
+      res.status(200).send(message);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(400).send(`Unable to delete rentals id ${id}`);
+    });
 });
 
+app.post("/api/rentals", (req, res) => {
+  const newRental = req.body;
+
+  Rental.find({})
+    .sort({_id: -1})
+    .limit(1)
+    .then(lastRental => {
+      newRental._id = lastRental[0]._id + 1;
+      Rental.create(newRental)
+    })
+    .then(results => {
+      res.status(200).send('New Rental added');
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(400).send('Unable to add new rental');
+    });
+});
+
+app.put("/api/rentals", (req, res) => {
+  const id = parseInt(req.query.id);
+  const update = req.body;
+
+  Rental.findOneAndUpdate({ _id: id }, update)
```

Comment:
findoneandupdate is more taxing than just update. I don't know if you need need the result returned, but if not this could benefit from being changed.

Reply:
Thanks for pointing that out. This could be something good to look into when it comes time to start optimizing.